cypress/e2e/custom/ecospheres/topics/elements/deeplink.cy.ts

```diff
@@ -0,0 +1,186 @@
+import type { Factor, Topic } from '@/model/topic'
+import {
+  factorFactory,
+  setupTopicWithExistingFactors,
+  visitTopic
+} from '../support'
+
+const VIEWPORT_WAIT = 500 // ms - time to wait for viewport to stabilize
+
+describe('Topic Elements - Deep Linking', () => {
+  let testTopic: Topic
+  let testFactors: Factor[]
+
+  beforeEach(() => {
+    ;({ testTopic, testFactors } = setupTopicWithExistingFactors())
+  })
+
+  describe('Navigating to a specific factor via URL hash', () => {
+    it('should navigate to a factor in a group when hash is present in URL', () => {
+      const targetFactor = testFactors[0]
```

Comment:
Could some of the assertions below be true even if the feature doesn't work just because it's the first factor, and e.g. has a good chance to already be in the viewport, or first group is already expanded by default (if we were to change expansion rules)?